.github/workflows/scripts/npm_release_files/ts_serialize.d.ts

```diff
@@ -1,6 +1,16 @@
 declare module "@gamebridgeai/ts_serialize" {
+  /** to be implemented by external authors on their models  */
+  export interface TransformKey {
+    /** a function that will be called against
+   * every property key transforming the key
+   * with the provided function
+   */
+    tsTransformKey(key: string): string;
+  }
+
   /** Adds methods for serialization */
   export abstract class Serializable {
+    public tsTransformKey?(key: string): string;
```

Comment:
What would you think about having a decorator on the class instead of a method?

```
@SerializeDefaults({ transformKey: TransformKey })
class Foo extends Serializable { ... }
```

... I looked into decorator inheritance a bit and it might be more trouble than it's worth, but just putting it out there.

https://medium.com/@ttemplier/angular2-decorators-and-class-inheritance-905921dbd1b7

Reply:
oh boy, JS decorators are so wonky, class decorators run after the property decorators, so this would require a big refactor. I believe class decorators are called when the class is initialized where property decorators are called during the construction of the class. 

With how we make our map we would need to update the map on the fly when the class is used and I am not sold on doing that, we did try a class decorator in an earlier iteration of the project and quickly ran into issues. 

I also prefer the approach in the PR using a class function and interface. We add the interface (and declare it) too. get warnings on. implementation. As the function is optional we need the interface for these warnings. That said authors can incorrectly have a tsTransformKey function signature without errors. if it is not implemented in the class, but this is okay IMHO because we provide the interface with instructions on how to implement it. `(s:  string): string`